riscv/triggers.cc

```diff
@@ -212,7 +224,7 @@ std::optional<match_result_t> mcontrol_common_t::detect_memory_access_match(proc
     value &= 0xffffffff;
   }
 
-  if (simple_match(xlen, value)) {
+  if (simple_match(xlen, value) && (action != ACTION_DEBUG_EXCEPTION || !suppress_breakpoint_triggers(proc->get_state()))) {
```

Comment:
This long expression appears 3 times. How about a common method `allow_action(action)`? It might make sense to fold this into `suppress_breakpoint_triggers()`.